modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/publisher/webapp/lifecycle/management/DeprecateWebAppTestCase.java

```diff
@@ -0,0 +1,106 @@
+import static org.testng.Assert.assertTrue;
+
+public class DeprecateWebAppTestCase {
+
+    private static final String TEST_DESCRIPTION = "Verify deprecating a published web app";
+    private APPMPublisherRestClient appmPublisherRestClient;
+    private String appName = "DeprecateWebAppTestCase";
+    private String appVersion = "1.0.0";
+    private String context = "/" + appName;
+    private String trackingCode = "AM_" + appName;
+    private String backEndUrl;
+    private String appCreatorUserName;
+    private String appCreatorPassword;
+    private String uuid;
+
+
+    @BeforeClass(alwaysRun = true)
+    public void startUp() throws Exception {
+        AutomationContext appMServer = new AutomationContext(AppmTestConstants.APP_MANAGER,
+                                                             TestUserMode.SUPER_TENANT_ADMIN);
+        backEndUrl = appMServer.getContextUrls().getWebAppURLHttps();
+        appmPublisherRestClient = new APPMPublisherRestClient(backEndUrl);
+
+        User appCreator = appMServer.getSuperTenant().getTenantUser("AppCreator");
+        appCreatorUserName = appCreator.getUserName();
+        appCreatorPassword = appCreator.getPassword();
+        appmPublisherRestClient.login(appCreator.getUserName(), appCreator.getPassword());
+
+        // Web app is created by AppCreator user.
+        HttpResponse appCreateResponse = appmPublisherRestClient.webAppCreate(appName, context,
+                                                                              appVersion,
+                                                                              trackingCode,
+                                                                              appCreatorUserName);
+        JSONObject appCreateResponseData = new JSONObject(appCreateResponse.getData());
+        uuid = appCreateResponseData.getString(AppmTestConstants.ID);
+        appmPublisherRestClient.changeState(uuid,
+                                            AppmTestConstants.LifeCycleStatus.SUBMIT_FOR_REVIEW);
+        appmPublisherRestClient.logout();
+
+        User appPublisher = appMServer.getSuperTenant().getTenantUser("AppPublisher");
+        appmPublisherRestClient.login(appPublisher.getUserName(), appPublisher.getPassword());
+
+        // Approving and publishing the web app.
+        appmPublisherRestClient.approveAndPublishWebApp(uuid);
+    }
+
+    @Test(description = TEST_DESCRIPTION)
+    public void testDeprecateWebApp() throws Exception {
+        //Deprecate the published web app.
+        HttpResponse httpResponse =
+                appmPublisherRestClient.changeState(uuid,
+                                                    AppmTestConstants.LifeCycleStatus.DEPRECATE);
+
+        int httpResponseCode = httpResponse.getResponseCode();
+        assertTrue(httpResponseCode == 200, httpResponseCode + " status code received.");
+        JSONObject approvedResponseData = new JSONObject(httpResponse.getData());
```

Comment:
Since this is a json response use the verifyJsonResponse method of VerificationUtil.java. This method will verify the status code to be 200, and content type to be application/json and payload to be a json object/array. 
Currently content type verification is commented because api fixes on content type by sajith is not committed yet